File: src/485.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
   public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int retVal = 0;

        int count = 0;
        for (int num : nums) {
            if (num == 1) {
                ++count;
            } else {
                retVal = max(retVal, count);
                count = 0;
            }
        }
        retVal = max(retVal, count);

        return retVal;
    }
};

int main(int argc, char** argv) {
    struct subject {
        vector<int> nums;
    };
    vector<subject> testData{{{1, 1, 0, 1, 1, 1}}, {{1, 0, 1, 1, 0, 1}}};
    int numberOfTestCase = testData.size();
    /* Example
     *  Input: nums = [1,1,0,1,1,1]
     *  Output: 3
     *
     *  Input: nums = [1,0,1,1,0,1]
     *  Output: 2
     */

    Solution cSolution;
    int answer;
    for (int i = 0; i < numberOfTestCase; ++i) {
        cout << "Input: nums = [";
        for (long unsigned int j = 0; j < testData[i].nums.size(); ++j) {
            cout << ((j == 0) ? "" : ",") << testData[i].nums[j];
        }
        cout << "]\n";

        answer = cSolution.findMaxConsecutiveOnes(testData[i].nums);
        cout << "Output: " << answer << "\n";

        cout << "\n";
    }

    return EXIT_SUCCESS;
}

File: src/485.py
import sys
import logging
from typing import List, Optional
from collections import defaultdict, deque, Counter
from functools import cache, lru_cache
from bisect import bisect_left, bisect_right
from itertools import accumulate
from heapq import heapify, heappush, heappop


def logging_setting():
    LOG_LEVEL = logging.INFO    # DEBUG, INFO, WARNING, ERROR, CRITICAL
    if __debug__:
        LOG_LEVEL = logging.DEBUG

    logging.basicConfig(
        stream=sys.stderr,
        level=LOG_LEVEL,
        format="%(levelname)-6s %(asctime)s %(module)s:%(funcName)s:%(lineno)-4s %(message)s",
        datefmt='%Y/%m/%d %H:%M:%S')


class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        retVal = 0

        count = 0
        for num in nums:
            if num == 1:
                count += 1
            else:
                retVal = max(retVal, count)
                count = 0
        retVal = max(retVal, count)

        return retVal


if __name__ == "__main__":
    logging_setting()

    try:
        logging.info("sys.version: %s", sys.version)
        print()

        pSolution = Solution()
        for nums in [[1, 1, 0, 1, 1, 1], [1, 0, 1, 1, 0, 1]]:
            # /* Example
            #  *  Input: nums = [1,1,0,1,1,1]
            #  *  Output: 3
            #  *
            #  *  Input: nums = [1,0,1,1,0,1]
            #  *  Output: 2
            #  */
            logging.info("Input: nums = %s", nums)

            retVal = pSolution.findMaxConsecutiveOnes(nums)
            logging.info("Output: %s", retVal)

            print()
    except KeyboardInterrupt as exception:
        logging.error("%s: %s", exception.__class__.__name__,
                      exception, exc_info=True)
        pass
    except Exception as exception:
        logging.error("%s: %s", exception.__class__.__name__,
                      exception, exc_info=True)
        pass

    sys.exit(0)
